src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/LoopingMessageProcessorServiceImplTest.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.service.impl;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.consumer.MessageConsumer;
+
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class LoopingMessageProcessorServiceImplTest {
+
+    @Mock
+    private ApplicationLogger logger;
+
+    @Mock
+    private MessageConsumer messageConsumer;
+
+    @InjectMocks
+    private LoopingMessageProcessorServiceImpl loopingMessageProcessorService;
+
+    @Test
+    void loopReadAndProcess() {
+        doAnswer(invocation -> {
+            ReflectionTestUtils.setField(loopingMessageProcessorService, "isRunning", false);
+            return null;
+        }).when(messageConsumer).readAndProcess();
+
+        var loopingMessageProcessorServiceResult =
+                loopingMessageProcessorService.loopReadAndProcess();
+
+        verify(messageConsumer, times(1)).readAndProcess();
+    }
```

Comment:
you might also be able to test that preDestroy sets isRunning to false by asserting the value after the call using ReflectionUtils

Reply:
i mean have a testPreDestroy that checks isRunning using reflection after method call